### DISCUSSION TITLE
target statistics

### DISCUSSION THREAD

**Discussion by thednainus** [EpiModel/EpiModel#610] [NetworkMethods] [answered]
I am not sure whether here is the right place to ask this question:
I am trying to estimate the target statistics for my model.
I only have the information from a paper which says:

Data were collected in 3 different time points:
Number of total respondents = 299
Number of sexual partners = 699 (I'm assuming this are the alters)
Median number of male sexual partners = 6  [IQR = 2 - 11] (I'm assuming this are the ties)

Median partnership duration = 0.79 months [IQR = 0.03, 6.97]

Is it correct to say that my `mean degree = (2 * 6)/(299+699)`

So edges will then be `mean degree * n/2`
Where `n` is the number of individuals at the beginning of my simulation?

**Comment by sgoodreau** [comment]
Hello - 

There are a few pieces here that are hard to follow or don't seem to be consistent.  I think it is a function of the wording you are using in your summary, relative to what the full questions actually asked are, and what it is your model is intended to capture.

Remember that mean degree as typically defined in target stats for a tergm measures the mean number of partner a person in the network has in the cross-section, ie at a moment in time.

You mention 299 respondents, who say they collectively they had 699 partners.  Are those the partners they had over the course of some time period?  Or the partners they have ongoing on the day of study?

If the latter, it would suggest that mean degree is 699/299 = 2.34. If the former, then you don't have enough info to calculate mean degree from just those two numbers.

Then you say that median number of male partners is 6.  But if the 299 people had 699 partners total, their male partners are presumably a subset of that.  How could they then have a median of 5?  Something in there does not add up.  I am guessing the questions are asking about different things that you haven't told us about.

In either case, your mean degree calculation is not right : it should have partners in the numerator somewhere, and the people having those partners in the denominator somewhere.  6 is already normalized by number of partners, and 699 is also about number of partners but is in the denominator instead of the numerator. 

Translating meandegree into edges is indeed edges = mean degree * n/2, where n is the population size used during model estimation (and also typically during the start of your subsequent network simulation).

**Comment by thednainus** [comment]
Hi Steven,

Thanks for the explanation. And yes, you correctly understood my question.
The paper describes a study which analysed data from 3 different _"waves"_ collected at different times. 
From these 3 dataset they had a total of 299 people that answer their questionnaire. These 299 individuals reported collectively 699 partners. 
And, respondents were asking about their partners in the past 3 months. So, that means I can't calculate the mean degree, right?

In this same paper they do report that the **median** number of sexual partners is 6 [IQR = 2.0 - 11].


I think I got confused on the calculation of the mean degree, after reading this tutorial http://statnet.org/Workshops/ergm.ego/ergm.ego_tutorial.html#Practical_issues
in which `mean degree = (2 x ties)/n`

Cheers,
Fabricia.

**Comment by sgoodreau** [comment]
For a right-tailed distribution (which typically includes sexual partners) the mean is larger than the median.  I don't really see a way for the reverse to be true for data like these.

One needs to be very careful about the details here.  For instance, was it set up where someone could answer a question about how many partners they had had total, and then they were also asked detailed partner-by-partner questions about a subset?  If so, the median of 6 may refer to the former, and the mean of 2.34 may refer to the latter.

In the end, our various tutorials and other materials can serve as a guide for helping people to learn how our methods and software works in reasonable detail, but every data situation is different, and users have to take their knowledge about the details of that data and adapt it as needed using first principles.  

On the workshop you link, to I see a place where it says `mean degree = (2 x ties)/n` but not one where it has nodes in the numerator.  Can you point me to where you see that?  If it says that we need to fix it.

**Comment by thednainus** [comment]
Sorry, it was a typo on my side, I meant _"ties"_ on the numerator and I've edited the post to fix it.

Back to mean and median, and based on your answer, I **don't think** from the paper, I would have enough information to calculate the mean degree because the number of partners were reported for the _past 3 months_.

Is there any tutorial that have some explanation to calculate the mean degree when respondents were asked to report the partners in the past XX months?
I don't have access to the respondents answers to the questionnaire. 
(Sorry, this is not really my area, and I am just looking for a way of calculating the mean degree based on the data described on this paper (if possible. it might be that the paper does not provide enough information). The idea is to use EpiModel to get a transmission matrix that I will convert to a phylogenetic tree).

Thanks,
Fabricia.

**Comment by sgoodreau** [comment, marked as answer]
Hi Fabricia - 

If all you have is mean of partners in the last XX months, then there is no way to calculate mean degree, since you don't have enough info - you need to know about mean relational duration too. Ignoring boundary/censoring effects for the moment and considering means of quantities only, the relationship is mean degree = mean duration * mean partner acquisition rate.  That is, if relationships last an average of 5 months, and people acquire an average of one new partner every 8 months (ie 1/8 new partner per month), then the mean degree will be 5/8.  Note that to calculate mean degree here, it doesn't matter the extent to which partnerships overlap or are sequential (ie what the momentary degree distribution is) - the mean will be 5/8.  Now, again ignoring boundary effects, you can think of the fact that folks acquire 1 new partner every 8 months as meaning that, they will average 1.5 new partners per year (12 months/8 months), or X/8 new partners every X months. But of course, in the last X months, they won't have only had the partners that they've initiated in those months; they'll also have some that started before.  So the number would be closer to (X+5)/8, that is, the partners that started within 5 months of the measurement window. The exact value depends on your assumptions/data about the distribution of relational durations (ie how likely is it that a relationship started at any point in the 5 months prior to your window would make it into your window), and that can get complex if one has data on relational durations that one wants to make full use of. But the above approach should be good enough for many purposes.

If you don't have any info on relational length, then I think you'll either need a new data source or have to make some assumptions.

In terms of all of this being written down - it really isn't in any single clear place, mainly because folks tend to derive the basic logic independently given the specific features of their data, which always vary in subtle ways. We cover elements of it in the NME course (https://statnet.org/nme/, click on Course Materials), but I can't think of anywhere that has all of the pieces I laid out above in one place.  Indeed, I had to work through the logic in my head as I wrote to get there for you. I can see it would be useful to have the above description somewhere in our materials for sure.

Note, of course, that all of this is just to get mean momentary degree.  To get the momentary degree distribution (ie how many people are in 0, 1, 2, etc. partnerships at a moment in time), one needs additional data.  Our group focuses a lot on this distribution, considering how much the same momentary mean degree *and* cumulative mean degree and even cumulative degree distribution can yield very different epidemic dynamics for different momentary degree distributions. For that you need to either ask people about relationships ongoing at the time of survey/interview, or collect start and end dates. Therein lies a whole additional realm of measurement considerations, with some literature, a lot of it by my colleague Martina Morris.
